src/components/Auth/Login.jsx

```diff
@@ -1,7 +1,141 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import fb from '../../assets/images/facebook.png';
+import twitter from '../../assets/images/twitter.png';
+import google from '../../assets/images/google.png';
+import Button from '../common/Button/Button';
+import Input from '../common/Input/Input';
+import { loginUser, inputHandler, validateCredentials } from '../../redux/actions';
 
-const LoginComponent = () => (
-  <h1 style={{ background: 'blue', color: 'white' }}>Login Component</h1>
-);
+class LoginComponent extends Component {
+  signin = (e) => {
+    const {
+      credentials: { username, password },
+      validate,
+      login,
+    } = this.props;
+    validate({ username, password }).then((res) => {
+      if (res.message === 'Ok') {
+        login({ username, password });
+      }
+    });
+    e.preventDefault();
+  };
 
-export default LoginComponent;
+  handleInput = (e) => {
+    const { handleInput } = this.props;
+    handleInput({ field: e.target.name, value: e.target.value });
+  };
+
+  handleError = () => {
+    const { error } = this.props;
+    if (error.message) {
+      return error.message;
+    } if (error.data) {
+      return error.data.message;
+    }
+    return '';
+  };
+
+  render() {
+    const {
+      credentials: { username, password },
+    } = this.props;
+    const { submitting } = this.props;
```

Comment:
If `credentials: { username, password }` and `submitting` are both `this.props` why are they on different lines?